screens/UsingScreen.js

```diff
@@ -1,93 +1,129 @@
-import React, { Component } from "react";
+import React, { Component, useEffect } from "react";
 import { StyleSheet, Alert } from "react-native";
-import { connect } from 'react-redux'
+import { connect } from "react-redux";
 import { Actions } from "react-native-router-flux";
 import Timer from "../components/Timer/Timer";
 import { Container, Content, Header, View, Segment, Banner } from "../components/layout";
 import { Button, IconButton } from "../components/buttons";
-import { computeDistance } from '../utils'
+import { computeDistance } from "../utils";
 import { makeAlarmLog, getNumberOfAvailableResponders } from "../store/actions";
 
 const fredVictorCoordinates = {
   lat: 43.6536212,
   lng: -79.3751693
-}
+};
 
 // TODO: Remember to change this to 500 prior to Beta testing
-const MAXIMUM_DISTANCE = 1000000000 // meters
+const MAXIMUM_DISTANCE = 1000000000; // meters
 
+const UsingScreen = props => {
+  const { location, time, userId, token } = props;
+  const { timeRemaining } = time;
 
-class UsingScreen extends Component {
-  constructor(props) {
-    super(props);
-    this.startAlarm.bind(this);
-  }
+  const startAlarm = () => {
+    // Check if location has been set
+    console.log({ location });
+    if (
+      location === null ||
+      location.coords === null ||
+      (location.coords.lat !== 0 && location.coords.lng !== 0)
```

Comment:
This does a check on local location right? But local location is always set at loading or login screen.
When creating a help request, the backend pulls user location from the database. So right now it's possible for a user to have a local location set, but no location set in the database, and when they start the alarm the help request gets sent with no location attached

Reply:
Btw wouldn't the condition be `(location.coords.lat == 0 || location.coords.lng == 0)`?

Reply:
Should we set the location once we retrieve it at loading screen then? Regardless of what address the lat lon might refer to? @ad2969 @mxnyee

Reply:
Honestly idk why we need to keep track of two locations? sorta confusing